pkg/agent/tunnel/tunnel.go

```diff
@@ -36,7 +62,78 @@ func Setup(config *config.Node) error {
 		return err
 	}
 
-	wsURL := fmt.Sprintf("wss://%s/v1-k3s/connect", config.ServerAddress)
+	client, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return err
+	}
+
+	addresses := []string{config.ServerAddress}
+
+	endpoint, _ := client.CoreV1().Endpoints("default").Get("kubernetes", metav1.GetOptions{})
+	if endpoint != nil {
+		addresses = getAddresses(endpoint)
```

Comment:
I've got the code here, Can be a flag not to use the address from k8s that if not in HA mode?
ref: #727

```
the address from k8s is diff with the real, when k3s-server in docker, or k3s-server in nat.  and k3s-agent conns not in the same lan env.
```